Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/BaseToolWithRectangleShape.kt

```diff
@@ -0,0 +1,786 @@
+private const val MAXIMAL_TOOL_STROKE_WIDTH = 8f
+private const val DEFAULT_ROTATION_SYMBOL_DISTANCE = 20
+private const val DEFAULT_ROTATION_SYMBOL_WIDTH = 30
+private const val DEFAULT_MAXIMUM_BOX_RESOLUTION = 0f
+private const val CLICK_IN_BOX_MOVE_TOLERANCE = 10
+private const val DEFAULT_RECTANGLE_SHRINKING = 0
+private const val HIGHLIGHT_RECTANGLE_SHRINKING = 5
+private const val DEFAULT_ROTATION_ENABLED = false
+private const val DEFAULT_RESIZE_POINTS_VISIBLE = true
+private const val DEFAULT_RESPECT_MAXIMUM_BORDER_RATIO = true
+private const val DEFAULT_RESPECT_MAXIMUM_BOX_RESOLUTION = false
+private const val CLICK_TIMEOUT_MILLIS = 250
+private const val BUNDLE_BOX_WIDTH = "BOX_WIDTH"
+private const val BUNDLE_BOX_HEIGHT = "BOX_HEIGHT"
+private const val BUNDLE_BOX_ROTATION = "BOX_ROTATION"
+
+abstract class BaseToolWithRectangleShape(
+    contextCallback: ContextCallback,
+    toolOptionsViewController: ToolOptionsVisibilityController,
+    toolPaint: ToolPaint,
+    workspace: Workspace,
+    commandManager: CommandManager
+) : BaseToolWithShape(
+    contextCallback,
+    toolOptionsViewController,
+    toolPaint,
+    workspace,
+    commandManager
+) {
+    private val rotationArrowArcStrokeWidth: Int
+    private val rotationArrowArcRadius: Int
+    private val rotationArrowHeadSize: Int
+    private val rotationArrowOffset: Int
+    private val arcPaint: Paint
+    private val arrowPaint: Paint
+    private val arcPath: Path
+    private val arrowPath: Path
+    private val tempDrawingRectangle: RectF
+    private val tempToolPosition: PointF
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    @JvmField
+    var boxWidth: Float
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    @JvmField
+    var boxHeight: Float
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    @JvmField
+    var boxRotation = 0f // in degree
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    @JvmField
+    var rotationEnabled: Boolean
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    @JvmField
+    var drawingBitmap: Bitmap? = null
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    @JvmField
+    var rotationSymbolDistance: Float? = 0f
+
+    @JvmField
+    var toolStrokeWidth: Float? = null
+
+    @JvmField
+    var maximumBoxResolution: Float
+
+    @JvmField
+    var resizePointsVisible: Boolean
+
+    @JvmField
+    var respectMaximumBorderRatio: Boolean
+
+    @JvmField
+    var respectMaximumBoxResolution: Boolean
+
+    @JvmField
+    var rectangleShrinkingOnHighlight: Int
+
+    private var boxResizeMargin: Float? = 0f
+    private var rotationSymbolWidth: Float? = 0f
+    private var currentAction: FloatingBoxAction? = null
+    private var overlayDrawable: Drawable? = null
+    private var downTimer: CountDownTimer? = null
+    private var resizeAction: ResizeAction
+    private var rotatePosition: RotatePosition
```

Comment:
I think we can entirely remove the rotatePosition variable, since its never actually used and also the according RotatePosition enum

Reply:
hmm.. yeah I was also thinking the same while refactoring the file :)